### PR TITLE
Temporary

### PR DESCRIPTION
**Pull request by prateekdagurtech**

### CODE REVIEW

File: database/sequelize.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const Sequelize = require('sequelize')
+const sequelize = new Sequelize('taskDB', 'root', 'Node@123', {
```
Comment (vueexcel) [inline database/sequelize.js:2]: use `.env` for credentials

Comment (iamprashanttyagi) [inline database/sequelize.js]: use `.env` for credentials

File: models/user.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+const User = sequelize.define("users", {
+    firstname: {
+        type: Sequelize.STRING(20),
+        allowNull: false
+
+    },
+    lastname: {
+        type: Sequelize.STRING(20),
+        allowNull: false
+
```
Comment (iamprashanttyagi) [inline models/user.js]: remove all extra new lines

File: routes/users.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        res.send(createUser)
+    }
+
+    catch (err) {
+        res.status(500).send(err.message)
+    }
+})
+router.post('/address', userAuthentication.auth, async (req, res) => {
+    try {
+        console.log(req.user.user_id);
```
Comment (iamprashanttyagi) [inline routes/users.js]: remove all debug statement

File: routes/users.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+router.get('/address', userAuthentication.auth, async (req, res) => {
+    try {
+        user_id = req.user.user_id
+        var userData = await UsersAddress.findAll({
+            include: [
+                {
+                    model: UsersModel, as: "users"
+                }
+            ]
+        });
```
Comment (iamprashanttyagi) [inline routes/users.js]: make array of address in the details of the user instead of user array in address

File: middleware/auth.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+const UsersAccessToken = require('../models/access_token');
+const Secret_Token = process.env.SECRET_TOKEN
+module.exports.auth = async (req, res, next) => {
+    try {
+        const token = req.headers.token
+        const decoded = jwt.verify(token, Secret_Token)
+        const user = await UsersAccessToken.findOne({ where: { access_token: token } })
+        console.log(user)
+        req.user = user
+        next()
```
Comment (iamprashanttyagi) [inline middleware/auth.js]: firstly check that token is exist in headers or not ?

File: middleware/passport.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+findByCredentials = async (username, password) => {
+    const user = await UsersModel.findOne({ where: { "username": username } })
+    if (!user) {
+        throw new Error("No such username")
+    }
+    isMatch = await bcrypt.compare(password, user.password)
+    if (!isMatch) {
+        throw new Error("No any matches of password")
+    }
+    return user
```
Comment (iamprashanttyagi) [inline middleware/passport.js]: do error handling

File: index.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+require("dotenv").config();
+const express = require('express')
+const passport = require('passport')
+const bodyParser = require('body-parser')
+const userRouters = require('./routes/users')
+let app = express()
+require('./database/sequelize')
+const sequelize = require('./database/sequelize')
```
Comment (iamprashanttyagi) [inline index.js]: why same file required two times ?

File: routes/users.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        res.status(401).send(e)
+    }
+})
+router.get('/get/page', async function (req, res) {
+    try {
+        const limit = parseInt(req.query.limit) || 3
+        const page = parseInt(req.query.page_no) || 1
+        const skip = limit * (page - 1)
+
+        users = await UsersModel.findAll({ limit: limit, offset: skip });
```
Comment (iamprashanttyagi) [inline routes/users.js]: `users` is not defined

File: routes/users.js [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+    if (!user) {
+        throw new Error("No such username")
+    }
+    isMatch = await bcrypt.compare(password, user.password)
+    if (!isMatch) {
+        throw new Error("No any matches of password")
+    }
+    return user
+}
```
Comment (iamprashanttyagi) [inline routes/users.js]: same function is coded two times.
code this only one time

**Review by iamprashanttyagi — COMMENTED** [review]
- send proper response format
- break code in small part and increase code re-usability 
- do proper error handling

File: routes/users.js (hunk 9 above)
Comment (prateekdagurtech) [reply routes/users.js]: done

File: routes/users.js (hunk 8 above)
Comment (prateekdagurtech) [reply routes/users.js]: done

File: index.js (hunk 7 above)
Comment (prateekdagurtech) [reply index.js]: done

File: middleware/passport.js (hunk 6 above)
Comment (prateekdagurtech) [reply middleware/passport.js]: done

File: middleware/auth.js (hunk 5 above)
Comment (prateekdagurtech) [reply middleware/auth.js]: done

File: routes/users.js (hunk 3 above)
Comment (prateekdagurtech) [reply routes/users.js]: done

File: models/user.js (hunk 2 above)
Comment (prateekdagurtech) [reply models/user.js]: done

File: routes/users.js (hunk 4 above)
Comment (prateekdagurtech) [reply routes/users.js]: done

**Pull request merged**